code/parachain/frame/cosmwasm/src/lib.rs

```diff
@@ -298,7 +298,8 @@ pub mod pallet {
 
 		/// A way to convert from our native account to cosmwasm `Addr`.
 		type AccountToAddr: Convert<AccountIdOf<Self>, String>
-			+ Convert<String, Result<AccountIdOf<Self>, ()>>;
+			+ Convert<String, Result<AccountIdOf<Self>, ()>>
+			+ Convert<Vec<u8>, Result<AccountIdOf<Self>, ()>>;
 
```

Comment:
If something implements Convert<String>, doesn't it always implement Convert<Vec<u8>>, since String is a Vec<u8>?

Reply:
But the thing is string representation and the binary representation is different in this case:
eg. String representation of an address can be both `SS58` address or hex address. But the Vec is the canonical binary representation which means `String::as_bytes() != Vec`. So the implementor should provide how we can get an account id from the canonical binary representation.
But I think providing some comments is useful for people to see the rationale behind this, because you normally think "oh isn't string a vec at the end". Good catch.